File: src/main/java/net/golem/raknet/protocol/connected/DisconnectionNotificationPacket.java
package net.golem.raknet.protocol.connected;

import net.golem.raknet.codec.PacketDecoder;
import net.golem.raknet.codec.PacketEncoder;
import net.golem.raknet.protocol.DataPacket;
import net.golem.raknet.protocol.RakNetPacketIds;

public class DisconnectionNotificationPacket extends DataPacket {

	public DisconnectionNotificationPacket() {
		super(RakNetPacketIds.DISCONNECTION_REQUEST);
	}

	@Override
	public void encode(PacketEncoder encoder) {}

	@Override
	public void decode(PacketDecoder decoder) {}
}

File: src/main/kotlin/raknet/packet/protocol/connected/DisconnectionNotificationPacket.kt
package raknet.packet.protocol.connected

import raknet.packet.DataPacket
import raknet.packet.PacketType

class DisconnectionNotificationPacket(): DataPacket(PacketType.DISCONNECTION_NOTIFICATION.id()) {

    override fun encodeOrder(): Array<Any> {
        return arrayOf()
    }
}